File: week03/배소연/SWEA_9940_순열1.py
T = int(input())

for tc in range(1, T + 1):
    N = int(input()) # 주어지는 정수의 개수
    num_arr = list(map(int, input().split())) # 수열의 원소
    is_ok = True
    
    if len(set(num_arr)) == N:
        
        for num in num_arr:
            if 0 <= num <= N:
                is_ok = True
            else:
                is_ok = False
    else:
        is_ok = False
        
        
    if is_ok:
        print(f'#{tc} Yes')
    else:
        print(f'#{tc} No')

File: week03/배소연/SWEA_9940_순열1.java
import java.io.*;
import java.util.*;

public class Solution {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int T = Integer.parseInt(br.readLine());

        for (int tc = 1; tc <= T; tc++) {
            int N = Integer.parseInt(br.readLine());

            StringTokenizer st = new StringTokenizer(br.readLine());

            Set<Integer> set = new HashSet<>();
            boolean isOk = true;

            for (int i = 0; i < N; i++) {
                int num = Integer.parseInt(st.nextToken());

                if (num < 1 || num > N) {
                    isOk = false;
                }

                if (!set.add(num)) {
                    isOk = false;
                }
            }

            if (isOk) {
                System.out.println("#" + tc + " Yes");
            } else {
                System.out.println("#" + tc + " No");
            }
        }
    }
}